cannot change scala.js version to 1.8.0

after pasting command:

```
$scala-cli --js --js-version=1.8.0 ./examples/scala-js/foojs.sc 
```
such exception is thrown:

```
Compiling project (Scala 3.0.2, Scala.JS)
Compiled project (Scala 3.0.2, Scala.JS)
Exception in thread "main" org.scalajs.ir.IRVersionNotSupportedException: Failed to deserialize a file compiled with Scala.js 1.8 (supported up to: 1.7): /home/slovic/.cache/coursier/v1/https/repo1.maven.org/maven2/org/scala-js/scalajs-library_2.13/1.8.0/scalajs-library_2.13-1.8.0.jar:/scala/volatile.sjsir
	at org.scalajs.linker.interface.unstable.IRFileImpl$$anonfun$withPathExceptionContext$1.applyOrElse(IRFileImpl.scala:62)
	at org.scalajs.linker.interface.unstable.IRFileImpl$$anonfun$withPathExceptionContext$1.applyOrElse(IRFileImpl.scala:58)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
	at scala.util.Failure.recover(Try.scala:234)
	at scala.concurrent.Future.$anonfun$recover$1(Future.scala:395)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:567)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
Caused by: org.scalajs.ir.IRVersionNotSupportedException: This version (1.8) of Scala.js IR is not supported. Supported versions are up to 1.7
	at org.scalajs.ir.VersionChecks.checkSupported(ScalaJSVersions.scala:77)
	at org.scalajs.ir.Serializers$Deserializer.readHeader(Serializers.scala:1010)
	at org.scalajs.ir.Serializers$Deserializer.deserializeEntryPointsInfo(Serializers.scala:969)
	at org.scalajs.ir.Serializers$.$anonfun$deserializeEntryPointsInfo$1(Serializers.scala:52)
	at org.scalajs.ir.Serializers$.withBigEndian(Serializers.scala:67)
	at org.scalajs.ir.Serializers$.deserializeEntryPointsInfo(Serializers.scala:52)
	at org.scalajs.linker.standard.MemIRFileImpl.$anonfun$entryPointsInfo$1(MemIRFile.scala:31)
	at org.scalajs.linker.standard.MemIRFileImpl.$anonfun$withBuffer$1(MemIRFile.scala:39)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	... 11 more
```

It seams that scala version doesn't matter. 
I guess it is also not important right now.

--- a/build.sc
+++ b/build.sc
@@ -364,8 +364,6 @@ trait Core extends ScalaCliSbtModule with ScalaCliPublishModule with HasTests
          |  def scalajsEnvJsdomNodejsVersion = "${Deps.scalaJsEnvJsdomNodejs.dep.version}"
          |  def scalaNativeVersion = "${Deps.nativeTools.dep.version}"
          |
-         |  def scalaJsCliVersion = "${InternalDeps.Versions.scalaJsCli}"
-         |
          |  def testRunnerOrganization = "$testRunnerOrganization"
          |  def testRunnerModuleName = "${`test-runner`(Scala.runnerScala3).artifactName()}"
          |  def testRunnerVersion = "${`test-runner`(Scala.runnerScala3).publishVersion()}"
@@ -704,7 +702,6 @@ trait Cli extends SbtModule with ProtoBuildModule with CliLaunchers
          |object Constants {
          |  def defaultScalaVersion = "${Scala.defaultUser}"
          |  def scalaJsVersion = "${Scala.scalaJs}"
-         |  def scalaJsCliVersion = "${InternalDeps.Versions.scalaJsCli}"
          |  def scalaNativeVersion = "${Deps.nativeTools.dep.version}"
          |  def ammoniteVersion = "${Deps.ammonite.dep.version}"
          |  def defaultScalafmtVersion = "${Deps.scalafmtCli.dep.version}"

--- a/modules/cli/src/main/scala/scala/cli/commands/shared/ScalaJsOptions.scala
+++ b/modules/cli/src/main/scala/scala/cli/commands/shared/ScalaJsOptions.scala
@@ -80,7 +80,7 @@ final case class ScalaJsOptions(
   @Hidden
     jsLinkerPath: Option[String] = None,
   @Group(HelpGroup.ScalaJs.toString)
-  @HelpMessage(s"Scala.js CLI version to use for linking (${Constants.scalaJsCliVersion} by default).")
+  @HelpMessage(s"Scala.js CLI version to use for linking (${Constants.scalaJsVersion} by default).")
   @ValueDescription("version")
   @Tag(tags.implementation)
   @Hidden

--- a/modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala
+++ b/modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala
@@ -232,6 +232,7 @@ final case class SharedOptions(
       linkerPath = jsLinkerPath
         .filter(_.trim.nonEmpty)
         .map(os.Path(_, Os.pwd)),
+      scalaJsVersion = jsVersion.map(_.trim).filter(_.nonEmpty),
       scalaJsCliVersion = jsCliVersion.map(_.trim).filter(_.nonEmpty),
       javaArgs = jsCliJavaArg,
       useJvm = jsCliOnJvm.map {

--- a/modules/cli/src/main/scala/scala/cli/internal/ScalaJsLinker.scala
+++ b/modules/cli/src/main/scala/scala/cli/internal/ScalaJsLinker.scala
@@ -36,15 +36,8 @@ object ScalaJsLinker {
       case None =>
         val scalaJsCliVersion = options.finalScalaJsCliVersion
         val scalaJsCliDep = {
-          val mod =
-            if (scalaJsCliVersion.contains("-sc"))
-              if (Version(scalaJsCliVersion) < Version("1.1.2-sc1"))
-                mod"io.github.alexarchambault.tmp:scalajs-cli_2.13"
-              else
-                mod"io.github.alexarchambault.tmp:scalajscli-${scalaJsVersion}_2.13"
-            else
-              mod"org.scala-js:scalajs-cli_2.13"
-          dependency.Dependency(mod, scalaJsCliVersion)
+          val mod = mod"org.virtuslab.scala-cli:scalajscli_2.13"
+          dependency.Dependency(mod, s"$scalaJsCliVersion+")
         }
 
         val forcedVersions = Seq(
@@ -82,11 +75,11 @@ object ScalaJsLinker {
             command.flatMap(_.value)
 
           case Left(osArch) =>
-            val useLatest = scalaJsCliVersion == "latest"
+            val useLatest = scalaJsVersion == "latest"
             val ext       = if (Properties.isWin) ".zip" else ".gz"
             val tag       = if (useLatest) "launchers" else s"v$scalaJsCliVersion"
             val url =
-              s"https://github.com/scala-cli/scala-js-cli-native-image/releases/download/$tag/scala-js-ld-$scalaJsVersion-$osArch$ext"
+              s"https://github.com/virtusLab/scala-js-cli/releases/download/$tag/scala-js-ld-$osArch$ext"
             val params = ExternalBinaryParams(
               url,
               useLatest,

--- a/modules/integration/src/test/scala/scala/cli/integration/ExportJsonTestDefinitions.scala
+++ b/modules/integration/src/test/scala/scala/cli/integration/ExportJsonTestDefinitions.scala
@@ -193,10 +193,10 @@ abstract class ExportJsonTestDefinitions(val scalaVersionOpt: Option[String])
       val fileContents = readJson(root / "dest" / "export.json")
 
       expect(fileContents ==
-        """{
+        s"""{
           |"scalaVersion": "3.1.3",
           |"platform": "JS",
-          |"scalaJsVersion": "1.12.0",
+          |"scalaJsVersion": "${Constants.scalaJsVersion}",
           |"jsEsVersion":"es2015",
           |"scopes": {
           | "main": {

--- a/modules/options/src/main/scala/scala/build/Artifacts.scala
+++ b/modules/options/src/main/scala/scala/build/Artifacts.scala
@@ -169,18 +169,13 @@ object Artifacts {
           fetched.fullDetailedArtifacts.collect { case (_, _, _, Some(f)) => os.Path(f, Os.pwd) }
 
         val scalaJsCliDependency =
-          scalaArtifactsParams.scalaJsCliVersion.map { version =>
-            val scalaJsVersion =
-              scalaArtifactsParams.scalaJsVersion.getOrElse(Constants.scalaJsVersion)
-            val mod =
-              if (version.contains("-sc"))
-                Module(
-                  Organization("io.github.alexarchambault.tmp"),
-                  ModuleName(s"scalajscli-${scalaJsVersion}_2.13"),
-                  Map.empty
-                )
-              else cmod"org.scala-js:scalajs-cli_2.13"
-            Seq(coursier.Dependency(mod, version))
+          scalaArtifactsParams.scalaJsCliVersion.map { scalaJsCliVersion =>
+            val mod = Module(
+              Organization("org.virtuslab.scala-cli"),
+              ModuleName(s"scalajscli_2.13"),
+              Map.empty
+            )
+            Seq(coursier.Dependency(mod, s"$scalaJsCliVersion+"))
           }
 
         val fetchedScalaJsCli = scalaJsCliDependency match {

--- a/modules/options/src/main/scala/scala/build/options/BuildOptions.scala
+++ b/modules/options/src/main/scala/scala/build/options/BuildOptions.scala
@@ -344,7 +344,9 @@ final case class BuildOptions(
           scalaJsVersion =
             if (platform.value == Platform.JS) Some(scalaJsOptions.finalVersion) else None,
           scalaJsCliVersion =
-            if (platform.value == Platform.JS) Some(Constants.scalaJsCliVersion) else None,
+            if (platform.value == Platform.JS)
+              Some(notForBloopOptions.scalaJsLinkerOptions.finalScalaJsCliVersion)
+            else None,
           scalaNativeCliVersion =
             if (platform.value == Platform.Native) Some(scalaNativeOptions.finalVersion) else None,
           addScalapy =

--- a/modules/options/src/main/scala/scala/build/options/scalajs/ScalaJsLinkerOptions.scala
+++ b/modules/options/src/main/scala/scala/build/options/scalajs/ScalaJsLinkerOptions.scala
@@ -7,11 +7,12 @@ final case class ScalaJsLinkerOptions(
   javaArgs: Seq[String] = Nil,
   /** If right, use JVM, if left, use the value as architecture */
   useJvm: Option[Either[String, Unit]] = None,
+  scalaJsVersion: Option[String] = None,
   scalaJsCliVersion: Option[String] = None,
   linkerPath: Option[os.Path] = None
 ) {
-  def finalScalaJsCliVersion = scalaJsCliVersion.getOrElse {
-    Constants.scalaJsCliVersion
+  def finalScalaJsCliVersion = scalaJsCliVersion.orElse(scalaJsVersion).getOrElse {
+    Constants.scalaJsVersion
   }
 
   /** If right, use JVM, if left, use the value as architecture */

--- a/project/deps.sc
+++ b/project/deps.sc
@@ -74,7 +74,6 @@ object InternalDeps {
   object Versions {
     def mill          = os.read(os.pwd / ".mill-version").trim
     def lefouMillwRef = "166bcdf5741de8569e0630e18c3b2ef7e252cd96"
-    def scalaJsCli    = "1.1.4-sc2"
   }
 }
 

--- a/project/deps.sc
+++ b/project/deps.sc
@@ -14,7 +14,7 @@ object Scala {
   val mainVersions = Seq(scala3, scala213)
   val runnerScalaVersions = runnerScala3 +: allScala2
 
-  def scalaJs = "1.12.0"
+  def scalaJs = "1.13.0"
 
   def listAll: Seq[String] = {
     def patchVer(sv: String): Int =
@@ -74,7 +74,7 @@ object InternalDeps {
   object Versions {
     def mill          = os.read(os.pwd / ".mill-version").trim
     def lefouMillwRef = "166bcdf5741de8569e0630e18c3b2ef7e252cd96"
-    def scalaJsCli    = "1.1.3-sc1"
+    def scalaJsCli    = "1.1.4-sc2"
   }
 }
 

--- a/website/docs/guides/scala-js.md
+++ b/website/docs/guides/scala-js.md
@@ -179,11 +179,12 @@ In the future, the Scala CLI will be able to support any version of Scala.js ind
 
 The table below lists the last supported version of Scala.js in Scala CLI. If you want to use newer Scala.js, update scala-cli.
 
-| Scala CLI versions   |      Scala.js      |
-|----------------------|:------------------:|
-| 0.0.9                |   1.7.1            |
-| 0.1.0 - 0.1.2        |   1.8.0            |
-| 0.1.3                |   1.9.0            |
-| 0.1.4 - 0.1.8        |   1.10.0           |
-| 0.1.9 - 0.1.17       |   1.10.1           |
-| 0.1.18 - current     |   1.12.0           |
+| Scala CLI versions | Scala.js |
+|--------------------|:--------:|
+| 0.0.9              |  1.7.1   |
+| 0.1.0 - 0.1.2      |  1.8.0   |
+| 0.1.3              |  1.9.0   |
+| 0.1.4 - 0.1.8      |  1.10.0  |
+| 0.1.9 - 0.1.17     |  1.10.1  |
+| 0.1.18 - 0.2.0     |  1.12.0  |
+| 0.2.1 - current    |  1.13.0  |

--- a/website/docs/reference/cli-options.md
+++ b/website/docs/reference/cli-options.md
@@ -1185,7 +1185,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 ### `--js-version`
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 ### `--js-mode`
 
@@ -1245,7 +1245,7 @@ Path to the Scala.js linker
 ### `--js-cli-version`
 
 [Internal]
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 ### `--js-cli-java-arg`
 

--- a/website/docs/reference/scala-command/cli-options.md
+++ b/website/docs/reference/scala-command/cli-options.md
@@ -658,7 +658,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 `SHOULD have` per Scala Runner specification
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 ### `--js-mode`
 
@@ -746,7 +746,7 @@ Path to the Scala.js linker
 
 `IMPLEMENTATION specific` per Scala Runner specification
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 ### `--js-cli-java-arg`
 

--- a/website/docs/reference/scala-command/runner-specification.md
+++ b/website/docs/reference/scala-command/runner-specification.md
@@ -118,7 +118,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 **--js-version**
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 **--js-mode**
 
@@ -358,7 +358,7 @@ Path to the Scala.js linker
 
 **--js-cli-version**
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 **--js-cli-java-arg**
 
@@ -815,7 +815,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 **--js-version**
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 **--js-mode**
 
@@ -1039,7 +1039,7 @@ Path to the Scala.js linker
 
 **--js-cli-version**
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 **--js-cli-java-arg**
 
@@ -1336,7 +1336,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 **--js-version**
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 **--js-mode**
 
@@ -1566,7 +1566,7 @@ Path to the Scala.js linker
 
 **--js-cli-version**
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 **--js-cli-java-arg**
 
@@ -1883,7 +1883,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 **--js-version**
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 **--js-mode**
 
@@ -2123,7 +2123,7 @@ Path to the Scala.js linker
 
 **--js-cli-version**
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 **--js-cli-java-arg**
 
@@ -2449,7 +2449,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 **--js-version**
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 **--js-mode**
 
@@ -2689,7 +2689,7 @@ Path to the Scala.js linker
 
 **--js-cli-version**
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 **--js-cli-java-arg**
 
@@ -2991,7 +2991,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 **--js-version**
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 **--js-mode**
 
@@ -3213,7 +3213,7 @@ Path to the Scala.js linker
 
 **--js-cli-version**
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 **--js-cli-java-arg**
 
@@ -3570,7 +3570,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 **--js-version**
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 **--js-mode**
 
@@ -3812,7 +3812,7 @@ Path to the Scala.js linker
 
 **--js-cli-version**
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 **--js-cli-java-arg**
 
@@ -4188,7 +4188,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 **--js-version**
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 **--js-mode**
 
@@ -4406,7 +4406,7 @@ Path to the Scala.js linker
 
 **--js-cli-version**
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 **--js-cli-java-arg**
 
@@ -5001,7 +5001,7 @@ Enable Scala.js. To show more options for Scala.js pass `--help-js`
 
 **--js-version**
 
-The Scala.js version (1.12.0 by default).
+The Scala.js version (1.13.0 by default).
 
 **--js-mode**
 
@@ -5219,7 +5219,7 @@ Path to the Scala.js linker
 
 **--js-cli-version**
 
-Scala.js CLI version to use for linking (1.1.3-sc1 by default).
+Scala.js CLI version to use for linking (1.13.0 by default).
 
 **--js-cli-java-arg**
 

